models/actions/task_output.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package actions
+
+import (
+	"context"
+	"crypto/sha1"
```

Comment:
I guess sha1 shouldn't be used because it's deprecated.

Reply:
The result size of sha256 is a bit long. In most cases, the key is shorter than 64 chars, so it may not bring optimization.

Should I use MD5 or `sha256.Sum256()[:X]`? Or any other ideas?

Reply:
Maybe 5cad940496 ?

Reply:
I can see that the `hash` column is used for `unique`?

If most keys are short keys, I think using `ID` +  `Key` is better than `ID` + `KeyHash`, unless most keys are long keys (longer than the hash length).

For example:

* `ID` +  `Key`, then database stores:
    *  123, key-1
    *  123, key-2
    *  123, key-3 

* `ID` +  `KeyHash`, then database stores:
    *  123, (32 or 40 chars)
    *  123, (32 or 40 chars)
    *  123, (32 or 40 chars)


The `ID` +  `Key` seems more friendly to database indices if most keys are short.

Reply:
You are right, how stupid I was since I already know that most keys are shorter than hashes.

Fixed in 5519d9d87b